src/Arcus.Messaging.Tests.Unit/ServiceBus/IServiceCollectionExtensionsTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Threading.Tasks;
+using Arcus.Messaging.Pumps.ServiceBus;
+using Arcus.Security.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Arcus.Messaging.Tests.Unit.ServiceBus
+{
+    public class IServiceCollectionExtensionsTests
+    {
+        [Fact]
+        public async Task AddServiceBusTopicMessagePump_WithSubscriptionNameIndirectSecretProvider_WiresUpCorrectly()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            var spySecretProvider = new Mock<ISecretProvider>();
```

Comment:
Spy 😂

Reply:
😄 yeah, believe it or not; but that's the official terminology of xUnit (not the library but the pattern) testing practices 😃 
http://xunitpatterns.com/Test%20Spy.html